test/acceptance/features/unbindAppToService.feature

```diff
@@ -107,3 +107,50 @@ Feature: Unbind an application from a service
 
         Then The env var "BACKEND_HOST" is not available to the application
         And The env var "BACKEND_USERNAME" is not available to the application
+    
+    Scenario: Remove binding when application does not exist anymore
+        Given OLM Operator "backend" is running
+        * The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: Backend
+            metadata:
+                name: example-backend
+                annotations:
+                    service.binding/host: path={.spec.host}
+                    service.binding/username: path={.spec.username}
+            spec:
+                host: example.com
+                username: foo
+            """
+        * Generic test application "generic-app-a-d-u" is running
+        * Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-a-d-u
+            spec:
+                application:
+                    name: generic-app-a-d-u
+                    group: apps
+                    version: v1
+                    resource: deployments
+                services:
+                -   group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: example-backend
+                    id: backend
+            """
+        * The application env var "BACKEND_HOST" has value "example.com"
+        * The application env var "BACKEND_USERNAME" has value "foo"
+
+        When App is deleted
+          """
+          apiVersion: apps/v1
+          kind: Deployment
+          metadata:
+            name: generic-app-a-d-u
+          """
+        Then Service binding "binding-request-a-d-u" is deleted
```

Comment:
The PR description describes a scenario where 

- if you delete sbr and ksvc at the same time, i.e.

```
kubectl  delete sbr binding-request-knative &
kubectl  delete ksvc knative-vcap &
```
which is not the case that is being demonstrated in the Test case here. Here, in this particular test case,  we are only deleting the app and according to the PR description that is not what the PR is resolving right? cc @cdlliuy

Reply:
@Avni-Sharma ,  yes, the test case in this PR added the missing piece for unbind check in previous acceptance test. 

But for the code fix of PR ..  it i hardly to be tested here since it only happens in race condition ...  given I run the cmd 
```
kubectl  delete sbr binding-request-knative &
kubectl  delete ksvc knative-vcap &
```
I need to run it multiple times to catch 1 or 2 occurrences ...  

I don't know how to do the race condition check in acceptance test.  Any idea?